maas/resource_maas_network_interface_link.go

```diff
@@ -188,12 +188,6 @@ func getNetworkInterfaceLinkParams(d *schema.ResourceData, subnetID int) *entity
 }
 
 func createNetworkInterfaceLink(client *client.Client, machineSystemID string, networkInterfaceID int, params *entity.NetworkInterfaceLinkParams) (*entity.NetworkInterfaceLink, error) {
-	// Clear existing links
```

Comment:
I am not convinced that this change is required while it solved the problem with your machines. Can we please revert it since it appears to be non relevant to your PR?

Reply:
I reverted the change and I will let acceptance tests determine if it was required